### ISSUE TITLE
In MySQL, an incorrect column default is generated when the default is CURRENT_TIMESTAMP

### ISSUE THREAD

**Issue by pettermahlen** [jOOQ/jOOQ#5574]
I'm trying to create a DB schema for a clean MySQL database instance used for integration tests, using this code:

```
    dslContext = DSL.using(connection, SQLDialect.MYSQL);
    dslContext.createTableIfNotExists(INSTANCES)
        .columns(HOST, PORT, DISCOVERY_NAME, ACTIVE, UPDATED_TIMESTAMP)
        .execute();
```

I get the following message:

```
org.jooq.exception.DataAccessException: SQL [create table if not exists `runtime_metadata`.`instances`(`host` varchar(255) not null, `port` int not null, `discovery_name` varchar(255) not null, `active` tinyint null default 1, `updated_timestamp` timestamp not null default null)]; Invalid default value for 'updated_timestamp'

...
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Invalid default value for 'updated_timestamp'
```

The table has been created via:

```
create table instances (
  host varchar(255) not null,
  port int not null,
  discovery_name varchar(255) not null,
  active tinyint default 1,
  updated_timestamp timestamp not null,
  PRIMARY KEY (host, port, discovery_name)
);
```

And the generated code for the field looks like:

```
    public final TableField<InstancesRecord, Timestamp> UPDATED_TIMESTAMP = createField("updated_timestamp", org.jooq.impl.SQLDataType.TIMESTAMP.nullable(false).defaultValue(org.jooq.impl.DSL.inline("CURRENT_TIMESTAMP", org.jooq.impl.SQLDataType.TIMESTAMP)), this, "");
```

Seems like the SQL generation is not picking up the correct default value? I tried explicitly setting the default to 0. but the SQL command issued still has 'null' as the default.

**Comment by lukaseder** [comment]
Thank you very much for your report. The problem is that the generated source code contains `DSL.inline()` here, when in this case it should be a `DSL.field()` call.

As a temporary workaround, you can replace the code in the generated sources (or patch the code generator to do so automatically).

Will investigate ASAP

**Comment by pronabb** [comment]
Will it be resolved in 3.11.0 version?

**Comment by lukaseder** [comment]
Yep. Fixed now. Thanks for the reminder.

**Comment by lukaseder** [comment]
Fix will be backported to 3.10.3 (#6891) and 3.9.7 (#6892)

**Comment by lukaseder** [comment]
This issue may have seen a regression in 3.16+, which will be fixed soon: https://github.com/jOOQ/jOOQ/issues/13818
